### ISSUE TITLE
When testing 204, aborted error occurs with Node 16

### ISSUE THREAD

**Issue by jeongmincha** [visionmedia/supertest#766]
My Environment
* Node 16.14.0
* @nestjs/core: 8.1.2
* @nestjs/platform-fastify: 8.2.6
* @nestjs/testing: 8.1.2
* jest: 27.0.6
* supertest: 6.2.2

My API code
```
@HttpCode(HttpStatus.NO_CONTENT)
@Get('/favicon.ico')
async favicon() {
  return null;
}
```

My test code
```
import {Test, TestingModule} from '@nestjs/testing';
import * as request from 'supertest';
import {AppModule} from '../../src/app.module';
import {FastifyAdapter, NestFastifyApplication} from '@nestjs/platform-fastify';

let app: NestFastifyApplication;

beforeAll(async () => {
  const moduleFixture: TestingModule = await Test.createTestingModule({
    imports: [AppModule],
  }).compile();

  app = moduleFixture.createNestApplication<NestFastifyApplication>(
    new FastifyAdapter()
  );
  await app.init();
  await app
    .getHttpAdapter()
    .getInstance()
    .ready();
});

it('/favicon (GET)', async () => {
  const response = await request(app.getHttpServer())
    .get('/favicon.ico')
    .expect(204);
});
```

My codes are successful when testing with node v12 (12.20.1), but it occurs `aborted()` error after I changed node v12 to v16.
```
Error: aborted

    at connResetException (node:internal/errors:691:14)
    at Socket.socketCloseListener (node:_http_client:402:19)
    at Socket.emit (node:events:532:35)
    at Socket.emit (node:domain:475:12)
    at TCP.<anonymous> (node:net:687:12)
    at TCP.callbackTrampoline (node:internal/async_hooks:130:17)
```

Only API endpoints returning 204 are making this kind of errors. Is there any way to solve it? I think it is related to supertest because my api server is working well when in dev/prod running mode, but it crashes only in testing 204 cases.

**Comment by titanism** [comment]
You may need to downgrade or reinstall due to a bad version of superagent which was just published and deprecated soon after.  Superagent is a dependency of supertest, try to `rm -rf node_modules/supertest` and reinstall.

**Comment by jeongmincha** [comment]
@titanism Thanks for answering quickly, but I'm using the latest version of sueprtest (6.2.2) and superagent (7.1.2). Is it not okay to use these versions even though these are the most latest?

**Comment by titanism** [comment]
superagent v7.1.2 is a bad version and was deprecated minutes ago

**Comment by jeongmincha** [comment]
@titanism I changed the versions to superagent (6.1.0) & supertest (6.1.2), but my error still occurs...

**Comment by jeongmincha** [comment]
I found the reason for this issue. 
In nest.js + node v16, we must not use `return null` with `@HttpCode(HttpStatus.NO_CONTENT)`, because it actually doesn't return empty response body. It returns the string `null`. I needed to change `return null` to `return ''` or `res.send()`.
In nest.js + node v12, I didn't encounter this issue, but it is known that we shouldn't return any response body with 204 API endpoints.
